modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/spi/model/query/contributor/DDMFormInstanceRecordModelPreFilterContributor.java

```diff
@@ -83,33 +80,22 @@ public void contribute(
 		if ((languageIds.length > 0) && (notEmptyFields.length > 0) &&
 			(structureId > 0)) {
 
-			List<Locale> locales = Stream.of(
-				languageIds
-			).map(
-				languageId -> LocaleUtil.fromLanguageId(languageId)
-			).collect(
-				Collectors.toList()
-			);
-
-			Stream.of(
-				notEmptyFields
-			).forEach(
-				notEmptyField -> {
-					BooleanFilter notEmptyFieldBooleanFilter =
-						new BooleanFilter();
-
-					locales.forEach(
-						locale -> notEmptyFieldBooleanFilter.add(
-							new ExistsFilter(
-								ddmIndexer.encodeName(
-									structureId, notEmptyField, locale)),
-							BooleanClauseOccur.MUST));
-
-					booleanFilter.add(
-						notEmptyFieldBooleanFilter,
-						BooleanClauseOccur.MUST_NOT);
+			for (String notEmptyField : notEmptyFields) {
+				BooleanFilter notEmptyFieldBooleanFilter = new BooleanFilter();
+
+				for (String languageId : languageIds) {
```

Comment:
@JanisZhang The `LocaleUtil.fromLanguageId` method looks a bit heavy. In the original logic, the `fromLanguageId` is called `languageIds.length` times; while in your logic, it's called `languageIds.length * notEmptyFields.length` times. In most occasions it's heavier than before. Please keep it a 2-step operation, thanks!